internal/commands/pull/command_test.go

```diff
@@ -317,6 +334,204 @@ Successfully pulled app down: app
 		assert.Nil(t, err)
 		assert.Equal(t, "<html><body>hello world!</body></html>", string(modifiedData))
 	})
+
+	t.Run("with a template ID to export with", func(t *testing.T) {
+
+		// Needs to have a successful export for templates
+		zipPkg, err := zip.OpenReader("testdata/test.zip")
+		assert.Nil(t, err)
+		defer zipPkg.Close()
+
+		var realmClient mock.RealmClient
+
+		realmClient.FindAppsFn = func(filter realm.AppFilter) ([]realm.App, error) {
+			return []realm.App{{ID: "appID", Name: "appName"}}, nil
+		}
+		realmClient.ExportFn = func(groupID, appID string, req realm.ExportRequest) (string, *zip.Reader, error) {
+			return "app_20210101", &zipPkg.Reader, nil
+		}
+
+		templateZipPkg1, err := zip.OpenReader("testdata/template_1.zip")
+		assert.Nil(t, err)
+		defer templateZipPkg1.Close()
+
+		templateZipPkg2, err := zip.OpenReader("testdata/template_2.zip")
+		assert.Nil(t, err)
+		defer templateZipPkg2.Close()
+
+		for _, tc := range []struct {
+			templateID       string
+			templateZip      *zip.Reader
+			expectedTemplate string
+		}{
+			{
+				templateID:  "template_1",
+				templateZip: &templateZipPkg1.Reader,
+				expectedTemplate: `{
+  "egg": "over easy",
+  "pancake": "blueberry",
+  "toast": "french"
+}`,
+			},
+			{
+				templateID:  "template_2",
+				templateZip: &templateZipPkg2.Reader,
+				expectedTemplate: `{
+  "egg": "scrambled",
+  "pancake": "plain",
+  "toast": "none"
+}`,
+			},
+		} {
+			t.Run(fmt.Sprintf("should export the corresponding template when the template ID %s is passed in", tc.templateID), func(t *testing.T) {
```

Comment:
[nit] avoid punctuation and capitalization in test descriptions

```suggestion
			t.Run(fmt.Sprintf("should export the corresponding template when the template id %s is passed in", tc.templateID), func(t *testing.T) {
```

also, [question]: why is this a test table?  i wanted to comment that we shouldn't need the actual assert value in the test description ("template id" here), and other than different template contents I don't understand what different logic these tests are proving.  could/should this be reduced down to one test?  seems like you'd still be able to prove the multiple templates were written

Reply:
I removed it! I was going to use it to test something else, but I ended up putting it somewhere else